### DISCUSSION TITLE
Too many calls to location or history APIs in a short period of time.

### DISCUSSION THREAD

**Discussion by mcolominas** [livewire/livewire#7746] [7. Bugs (archived)]
### Livewire version

3.3.5

### Laravel version

10.41.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Using Firefox, when you click several times on one or more links in the side menu (I'm using filament), the page goes blank. 

Checking the console, there is the following error:
Too many calls to location or history APIs in a short period of time.

I have tried it with chrome and this error does not occur, only in firefox.

What it makes me understand is that firefox, if `window.history.replaceState` is called several times, for security reasons, it gives an error and since livewire is not handling the errors (try cath), it causes the script to stop and the page cannot be loaded.

The expected thing would be for some modal or alert to appear or for it to be on the loading page and try it a few milliseconds later.

Example code, copy and paste it into firefox console:
```js
setInterval(function(){
history.replaceState( {} , 'foo', '/foo' );
}, 1)
```

### Repository or Wirebox URL

no url

### Additional information

Console log:

```
Too many calls to location or history APIs in a short period of time. livewire.js:7654:19
Uncaught DOMException: The operation is insecure.
    replace https://domain.com/livewire/livewire.js?id=e2b302e9:7654
    track2 https://domain.com/livewire/livewire.js?id=e2b302e9:7607
    <anonymous> https://domain.com/livewire/livewire.js?id=e2b302e9:8587
    <anonymous> https://domain.com/livewire/livewire.js?id=e2b302e9:8582
    trigger https://domain.com/livewire/livewire.js?id=e2b302e9:433
    processEffects https://domain.com/livewire/livewire.js?id=e2b302e9:4393
    Component https://domain.com/livewire/livewire.js?id=e2b302e9:4361
    initComponent https://domain.com/livewire/livewire.js?id=e2b302e9:4426
    start2 https://domain.com/livewire/livewire.js?id=e2b302e9:8268
    skipDuringClone https://domain.com/livewire/livewire.js?id=e2b302e9:1850
    initTree https://domain.com/livewire/livewire.js?id=e2b302e9:879
    initTree https://domain.com/livewire/livewire.js?id=e2b302e9:879
    walk https://domain.com/livewire/livewire.js?id=e2b302e9:802
    walk https://domain.com/livewire/livewire.js?id=e2b302e9:807
    walk https://domain.com/livewire/livewire.js?id=e2b302e9:807
    walk https://domain.com/livewire/livewire.js?id=e2b302e9:807
    walk https://domain.com/livewire/livewire.js?id=e2b302e9:807
    initTree https://domain.com/livewire/livewire.js?id=e2b302e9:877
    deferHandlingDirectives https://domain.com/livewire/livewire.js?id=e2b302e9:1360
    initTree https://domain.com/livewire/livewire.js?id=e2b302e9:876
    nowInitializeAlpineOnTheNewPage https://domain.com/livewire/livewire.js?id=e2b302e9:7551
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7502
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7543
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7542
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7498
    swapCurrentPageWithNewHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7321
    swapCurrentPageWithNewHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7320
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7489
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7540
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7485
    whenFinished https://domain.com/livewire/livewire.js?id=e2b302e9:7067
    storeThePrefetchedHtmlForWhenALinkIsClicked https://domain.com/livewire/livewire.js?id=e2b302e9:7053
    navigate_default https://domain.com/livewire/livewire.js?id=e2b302e9:7471
    prefetchHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7046
    promise callback*prefetchHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7045
    navigate_default https://domain.com/livewire/livewire.js?id=e2b302e9:7470
    whenThisLinkIsPressed https://domain.com/livewire/livewire.js?id=e2b302e9:7085
    whenThisLinkIsPressed https://domain.com/livewire/livewire.js?id=e2b302e9:7081
    navigate_default https://domain.com/livewire/livewire.js?id=e2b302e9:7468
    flushHandlers https://domain.com/livewire/livewire.js?id=e2b302e9:1353
    stopDeferring https://domain.com/livewire/livewire.js?id=e2b302e9:1358
    deferHandlingDirectives https://domain.com/livewire/livewire.js?id=e2b302e9:1361
    initTree https://domain.com/livewire/livewire.js?id=e2b302e9:876
    nowInitializeAlpineOnTheNewPage https://domain.com/livewire/livewire.js?id=e2b302e9:7551
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7502
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7543
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7542
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7498
    swapCurrentPageWithNewHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7321
    swapCurrentPageWithNewHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7320
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7489
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7540
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7485
    whenFinished https://domain.com/livewire/livewire.js?id=e2b302e9:7067
    storeThePrefetchedHtmlForWhenALinkIsClicked https://domain.com/livewire/livewire.js?id=e2b302e9:7053
    navigate_default https://domain.com/livewire/livewire.js?id=e2b302e9:7471
    prefetchHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7046
    promise callback*prefetchHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7045
    navigate_default https://domain.com/livewire/livewire.js?id=e2b302e9:7470
    whenThisLinkIsPressed https://domain.com/livewire/livewire.js?id=e2b302e9:7085
    whenThisLinkIsPressed https://domain.com/livewire/livewire.js?id=e2b302e9:7081
    navigate_default https://domain.com/livewire/livewire.js?id=e2b302e9:7468
    flushHandlers https://domain.com/livewire/livewire.js?id=e2b302e9:1353
    stopDeferring https://domain.com/livewire/livewire.js?id=e2b302e9:1358
    deferHandlingDirectives https://domain.com/livewire/livewire.js?id=e2b302e9:1361
    initTree https://domain.com/livewire/livewire.js?id=e2b302e9:876
    nowInitializeAlpineOnTheNewPage https://domain.com/livewire/livewire.js?id=e2b302e9:7551
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7502
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7543
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7542
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7498
    swapCurrentPageWithNewHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7321
    swapCurrentPageWithNewHtml https://domain.com/livewire/livewire.js?id=e2b302e9:7320
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7489
    preventAlpineFromPickingUpDomChanges https://domain.com/livewire/livewire.js?id=e2b302e9:7540
    navigateTo https://domain.com/livewire/livewire.js?id=e2b302e9:7485
```

```js
function replace(url, key, object) {
    let state = window.history.state || {};
    if (!state.alpine)
      state.alpine = {};
    state.alpine[key] = unwrap(object);
    window.history.replaceState(state, "", url.toString()); //this line
  }
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by binaryfire** [comment]
We're having the same issue. A couple of our customers using Firefox have reported this to us.

**Reply by bytemewebdev** [reply]
I'm experiencing the same only on Firefox.

**Comment by binaryfire** [comment]
Here's a screenshot. To reproduce, use `wire:navigate` and navigate between multiple pages in quick succession in Firefox:

<img width="555" alt="image" src="https://github.com/livewire/livewire/assets/5361908/7c73d31a-bfba-4960-a818-33de78e2dc0c">

**Comment by alkhatibsy** [comment]
I'm experiencing the same only on Firefox.

**Comment by binaryfire** [comment]
Not sure of the best way to bring this to the attention of the maintainers, because it's not something you can write a failing test for. And these discussions don't seem to be monitored.

**Reply by joshhanley** [reply]
@binaryfire thanks, I'm aware of it 🙂 not sure what to do about it yet though 🤔

**Reply by binaryfire** [reply]
@joshhanley Haha sorry, spoke too soon🙂

**Reply by joshhanley** [reply]
@binaryfire all good! 🙂 I've seen it pop up before and had a quick google but not looked into it beyond that. Will have a look further if I get a chance but I don't use firefox personally 😅

Definitely open to ideas if anyone has any!

**Reply by HDVinnie** [reply]
Im also facing this only in safari after updating from Livewire 2 to 3. 

https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/3608

Not using wire:navigate. 

<img width="1067" alt="Screen Shot 2024-03-04 at 2 06 41 PM" src="https://github.com/livewire/livewire/assets/12850699/6d0a9d5e-9850-4067-9c26-2c2012e9b854">

This causing alpine to not work as it should

**Reply by HDVinnie** [reply]
Ok for me this seems to happen if use to many #[Url] attributes. Like so [app/Http/Livewire/TorrentSearch.php](https://github.com/HDInnovations/UNIT3D-Community-Edition/blob/990f89c52825777432a1fdd549bce03a2d5f8334/app/Http/Livewire/TorrentSearch.php)

if removed works fine.

**Reply by davidwhicker** [reply]
> @binaryfire all good! 🙂 I've seen it pop up before and had a quick google but not looked into it beyond that. Will have a look further if I get a chance but I don't use firefox personally 😅
> 
> 
> 
> Definitely open to ideas if anyone has any!

Good day. I don't suppose you have had any luck here. I get the warning in Chrome and it disables navigate to other pages. https://issues.chromium.org/issues/40113103

**Comment by superuser-me** [comment]
I am facing the same issue in Filament (version 3.2.47),
when i refresh the page multiple times in short time span like 4-5 times in a 2 seconds,
it shows this error in console.
but after a second or two it starts working again,
looks like somekind of rate limit.
![CleanShot 2024-03-14 at 03 35 15 AM@2x](https://github.com/livewire/livewire/assets/71923103/3b42e11c-3b68-44bf-825a-23746fb78573)

**Reply by alisalehi1380** [reply]
hi bro @superuser-me 

I'm experiencing the same only on Firefox.
Did you find a solution to this problem?
Did you find a way?

**Reply by superuser-me** [reply]
@alisalehi1380 i think it's because of SPA mode in adminpanelprovider,
i have stopped using firefox and disabled SPA mode now this issue is gone.

**Reply by binaryfire** [reply]
@superuser-me The issue isn't Filament-related. It occurs when using `wire:navigate` in non-Filament projects too.

**Reply by alisalehi1380** [reply]
@superuser-me Yeah. This is an issue for SPA mode in Firefox. But I thought you solved the problem in spa mode. thanks🙏

@binaryfire Thank you for the explanation. 🙏

**Comment by nilskretschmer** [comment]
Also experiencing this issue with Firefox. I see no open issues in the livewire project here on GitHub. Should we open up an issue for this?

**Comment by GregoireGaonach** [comment]
I have exactly the same problem with Firefox. Chrome seems fine. I am not quite sure what is causing the problem. On my end, the problem is sporadic and doesn't seem to be specifically related to `wire:navigate`.

What I can say, however, is that it happens when the user tries to reload the same problematic page multiple times within a good 10s of each other. The page is quite heavy.

This is clearly beyond my knowledge, but I'm happy to sponsor any PR.

**Reply by mgkimsal** [reply]
FWIW, I'm seeing this in safari and Firefox on Mac.

On chrome, I don't see that same error, but do see

"Throttling navigation to prevent the browser from hanging. See https://crbug.com/1038223. Command line switch --disable-ipc-flooding-protection can be used to bypass the protection"

which seems related ('navigation')

**Reply by davidwhicker** [reply]
> FWIW, I'm seeing this in safari and Firefox on Mac.
> 
> 
> 
> On chrome, I don't see that same error, but do see
> 
> 
> 
> "Throttling navigation to prevent the browser from hanging. See https://crbug.com/1038223. Command line switch --disable-ipc-flooding-protection can be used to bypass the protection"
> 
> 
> 
> which seems related ('navigation')

I have this chrome error at the moment. Seems to stop me navigating to new pages for about 10 seconds. I don't have SPA/navigte mode on.

**Comment by davidwhicker** [comment]
I can confirm I am seeing this warning in Chrome: 
"Throttling navigation to prevent the browser from hanging. See https://crbug.com/1038223. Command line switch --disable-ipc-flooding-protection can be used to bypass the protection"

It stops me from navigating to new pages until about 10 seconds have passed. 

Its a heavy page that loads.

**Comment by lcamero** [comment]
Anyone been able to get around this? 

I noticed this issue in Safari while working on a FilamentPHP app, however, I was able to reproduce on a fresh Livewire project.

What I've found is that the #[Url] attribute getting called a lot of times (by rendering a component using it over and over) makes the error show. 

In my use case with Filament, I have a ParentComponent that loads X amount of ChildComponents. Those ChildComponents use the `InteractsWithActions` trait that has these 2 properties. I don't really make use of them directly but just by having them included in the trait on my ChildComponent creates the issue

```PHP
    /**
     * @var mixed
     */
    #[Url(as: 'action')]
    public $defaultAction = null;

    /**
     * @var mixed
     */
    #[Url(as: 'actionArguments')]
    public $defaultActionArguments = null;
```

So I created a fresh Livewire project, created 2 components (ParentComponent and ChildComponent) and basically loaded 100 ChildComponents.

ChildComponent.php
```PHP
<?php

namespace App\Livewire;

use Livewire\Component;
use Livewire\Attributes\Url;

class ChildComponent extends Component
{
    #[Url]
    public string $search;

    public function render()
    {
        return view('livewire.child-component');
    }
}
```

parent-component.blade.php
```
<div>
    @foreach (range(1, 100) as $item)
        <livewire:child-component />
    @endforeach
</div>
```

This throws the error (as long as you don't use #[Url(history: true)]. Now, this is not the best example because why would all of the ChildComponents be storing state in the url for "search"? 

Thing is that, going back to my original project, on Filament the actions have that trait configuring the #[Url] attribute, so when I render a handful of components in a list `<ul>` where each component has its actions (view, edit, etc) the error is thrown.
I'm assuming this is addressed somehow because in Filament you can have a table with N rows and each have their own actions, but I just haven't found that yet. 

This piece of code seems to help throttle the changes in the history state, but can certainly have unwanted side effects and does not ultimately address the issue
https://github.com/livewire/livewire/discussions/5923#discussioncomment-9202549

Hope someone has some input on this 🙏

**Comment by hugomyb** [comment]
Hi everyone!

I ran into the same problem on a project trying to display multiple livewire components in a loop.

```html
@foreach($tasks->whereNull('parent_id') as $task)
            <livewire:task-row :$task :$sortBy :key="'task-' . $task->id . '-' . Illuminate\Support\Str::uuid()"/>
@endforeach
```

I finally found a solution that works on my side. Just add this JS script to your view: 

```js
<script>
        const original = window.history.replaceState;
        let timer = Date.now();

        let timeout = null;
        let lastArgs = null;

        window.history.replaceState = function (...args) {
            const time = Date.now();
            if (time - timer < 300) {
                lastArgs = args;

                if (timeout) {
                    clearTimeout(timeout);
                }

                timeout = setTimeout(() => {
                    original.apply(this, lastArgs);

                    timeout = null;
                    lastArgs = null;
                }, 100);

                return;
            }

            timer = time;

            original.apply(this, args);
        };
    </script>
```
    
I hope it works for you too ! :smile:

**Reply by studgate** [reply]
that is working for me, I had wire:navigate in a few locations, thanks @hugomyb

**Comment by andrewdwallo** [comment]
I've been experiencing the same issue in a Livewire app (also using Filament) without SPA mode (`wire:navigate`), and after a few hours of some digging, I think identified the root cause. It seems related to how Livewire handles `#[Url]` attributes.

### The Problem

https://github.com/livewire/livewire/blob/main/src/Features/SupportQueryString/BaseUrl.php

Every time a page is loaded, Livewire's dehydrate hook calls the `pushQueryStringEffect` method, which automatically triggers `history.replaceState`, even if the query string has not changed or been modified. This happens unconditionally, regardless of whether any query strings are present or updated. When components with `#[Url]` attributes are re-rendered or loaded multiple times (e.g. Filament's `ListRecords` component), such as when navigating quickly, it results in too many calls to the browser's `replaceState`, leading to the "Too many calls to location or history APIs" error, particularly in browsers like Firefox and Safari.

### What’s Causing It

The issue stems from Livewire calling `history.replaceState` during every `dehydrate` phase, without checking if the query string has actually changed. This leads to unnecessary history state manipulations, overwhelming the browser’s limit for `replaceState` calls within a short timeframe.

### Possible Fix

The fix should involve adding a condition in Livewire’s frontend code, specifically in the `replace` function, to compare the current value in the URL’s query string with the new value before calling `replaceState`. If the values are the same, `replaceState` shouldn’t be called, which would prevent unnecessary browser API calls.

I'm not sure if this would also fix issues related to `wire:navigate` (SPA mode), but it's possible that this approach could help with that as well.

If someone with more experience in JavaScript and Livewire's frontend source code would like to take a shot at fixing this, it would be greatly appreciated!

**Reply by robinmalburn** [reply]
Just building on this reply, I did a little debugging on the project where I'm seeing this issue by wrapping the `History.replaceState` function and saw the following:

On one Filament resource listing, the Livewire calls `History.replaceState` 27 times, of which 26 of those calls are made with identical payloads (both the `url` and `state` being passed through are identical).

I'm currently using a wrapper to dedupe the History API calls, but I'd like to look into the actual Livewire source code to to see about trying to patch at source, but unfortunately getting time to set that up in a local docker container that plays nicely with Dusk/Selenium and whatnot is a bit of a challenge right now.

This is the wrapper script I'm using if it helps anyone else or if it helps anyone more able to quickly contribute to the project itself:

```
(function () {
    const original = window.history.replaceState;

    window.history.replaceState = function (state, unused, url) {
        state.url = url;

        if (JSON.stringify(state) === JSON.stringify(window.history.state)) {
            return;
        }

        original.apply(this, [state, unused, url]);
    };
})();
```

**Reply by andrewdwallo** [reply]
Hi @PhiloNL , I apologize for tagging you directly, but I thought it was the best way to bring this to your attention. I wasn’t sure whether you or Caleb would be the most appropriate person to address this.

This discussion post has been active for over 9 months, and my reply, which could be identifying the underlying issue, has been up for about a month. I believe someone with your experience in Livewire’s frontend code could resolve this fairly quickly given the information provided.

Fixing this would greatly improve the user and developer experience with both `wire:navigate` and applications that rely on query strings or the history API. I consider it a significant issue that would benefit from the maintainers' attention.

So far, 14 people have upvoted this post, and I've personally heard from many others who are facing the same issue.

**Reply by reppair** [reply]
We need this sorted. It is a deal breaker for me on the new project I am starting (probably for others too)! I'm considering a re-write with Inertia and Vue tbh.

**Reply by reppair** [reply]
Hey @robinmalburn I've tried your script, but it throws an `DOMException: URL object could not be cloned.` as soon as I click a link with `wire:navigate` or a button with `@click="Livewire.navigate('...')"` on it.

Turns out the `url` being assigned onto `state.url` is sometimes an instance of the URL object. So I had to add a check for this.
```js
    (function () {
        const original = window.history.replaceState;

        window.history.replaceState = function (state, unused, url) {
            state.url = url instanceof URL ? url.toString() : url

            if (JSON.stringify(state) === JSON.stringify(window.history.state)) {
                return
            }

            original.apply(this, [state, unused, url])
        };
    })()
```

**Reply by robinmalburn** [reply]
Good catch @reppair.  In my usage with Filament this hasn't cropped up, but it's good to know there's a risk of it.  Thanks for sharing the update, I'll merge it into my local fix just to be on the safe side even with Filament.

**Reply by andrewdwallo** [reply]
> We need this sorted. It is a deal breaker for me on the new project I am starting (probably for others too)! I'm considering a re-write with Inertia and Vue tbh.

Unfortunately, the PR that was made by @luanfreitasdev hasn't even been addressed yet. Seems like they either don't care at all about the issue, are ignoring it completely, or they are too busy with flux. Who knows..

**Reply by binaryfire** [reply]
Hi @joshhanley. Could you or Caleb please review https://github.com/livewire/livewire/pull/8987 

You mentioned you guys were aware of the problem in Feb but weren’t sure how to address is. Looks like there’s a solution now.

This is a major issue for a lot of people. I agree with Andrew - the radio silence isn’t great. Fixing bugs like this should be a higher priority than the commercial products and this has been open for a year now.

Until then, could you please add a “wire:navigate isn’t compatible with Firefox” notice to the docs? People should be aware of that when deciding on a frontend framework or they’ll end up in situations like this: https://github.com/livewire/livewire/discussions/7746#discussioncomment-11133167. Especially React / Vue devs who are new to Laravel since they'll naturally assume "SPA" functionality works with the main 4 browsers.

**Comment by robinmalburn** [comment]
Also seeing this issue in Firefox `130.0.1` on the latest MacOS.  Also initially found via Filament, but the error is originating in livewire.js as per everyone else.

**Comment by reppair** [comment]
Same on macOS `15.1 (24B83)` and Firefox Develiper Edition `133.0b3 (aarch64)`.

**Comment by binaryfire** [comment]
This has been problematic for a couple of our devops apps because most sysadmins use Firefox. Also, some IT firms only allow Firefox internally for security reasons. We've started transitioning one app to Inertia/Vue because our biggest customer has hit this bug several times now.

**Reply by reppair** [reply]
Yeah, as I stated above in a reply, I am considering the same. It is a pity! I love the stack but at the end of the day, if it doesn't work, it _does not work_. :(

**Comment by reppair** [comment]
@calebporzio sorry for tagging you directly here mate, but we would love to hear your thoughts on this. Cheers!

**Comment by luanfreitasdev** [comment]
https://github.com/livewire/livewire/pull/8987 😄

**Reply by CalumBThomson** [reply]
@luanfreitasdev thanks for this -- really very much need a fix for this! Does your PR address the underlying livewire issue, regardless of browser?

So not just for Firefox as suggested in the title, but also for Safari and even Chrome (which does not completely freeze up like Firefox and Safari, but does still throttle navigation when multiple calls are made in quick succession)?

**Reply by luanfreitasdev** [reply]
PR updated: https://github.com/livewire/livewire/pull/9059

@CalumBThomson this fixes it for all cases

**Comment by LA-87** [comment]
This fixed it for me. 

With other examples i had issues that URL query parameters were not updating.
```
#[Url]
public string $search;
```

```
const original = window.history.replaceState;
let previousState = null

window.history.replaceState = function (state, unused, url) {
    state.url = url instanceof URL ? url.toString() : url

    if (JSON.stringify(state) === JSON.stringify(previousState)) {
        return
    }

    original.apply(this, [state, unused, url])

    previousState = state
};
```

**Comment by bjuppa** [comment]
For those of you running into this problem with Livewire components using the Filament `InteractsWithActions` trait, it can be prevented by redeclaring the properties from the trait **without the `Url` attribute** in your component, like so:

```php
class YourComponent extends Component implements HasActions
{
    use InteractsWithActions;

    /**
     * Reset variable without Livewire's Url attribute added in InteractsWithActions trait.
     * Prevents issue with livewire navigation used by many components on same page
     * See https://github.com/livewire/livewire/discussions/5923
     * and https://github.com/livewire/livewire/discussions/7746
     *
     * @var mixed
     */
    public $defaultAction;

    /**
     * Reset variable without Livewire's Url attribute added in InteractsWithActions trait.
     *
     * @var mixed
     */
    public $defaultActionArguments;
```

Of course this only works if you don't rely on those properties being part of the url, but if you repeat a component multiple times on a page, that is probably not the case anyway!

But I agree with previous comments, that the best solution would be to have Livewire check the current value and skip setting the state if not changed.
